amundsen_application/static/js/components/common/TagsList/index.tsx

```diff
@@ -68,28 +41,36 @@ export class TagsList extends React.Component<TagsListProps, TagsListState> {
   }
 
   render() {
-    let innerContent;
     if (this.props.isLoading) {
-      innerContent = <LoadingSpinner/>;
-    } else {
-      innerContent = (
-        <div id="browse-body" className="browse-body">
-          {this.generateTagInfo(this.state.curatedTags)}
-          {
-            this.state.curatedTags.length > 0 && this.state.otherTags.length > 0 &&
-              <hr />
-          }
-          {this.generateTagInfo(this.state.otherTags)}
+      return <LoadingSpinner/>;
+    }
+    if (AppConfig.browse.curatedTags.length === 0) {
+      return (
+        <div id="tags-list" className="tags-list">
+          { this.generateTagInfo(this.props.allTags) }
         </div>
       );
     }
-    return (innerContent);
+    return (
+      <div id="tags-list" className="tags-list">
+        { this.generateTagInfo(this.props.curatedTags) }
+        {
+          AppConfig.browse.showAllTags && this.props.curatedTags.length > 0 && this.props.otherTags.length > 0 &&
```

Comment:
A utility method has the added benefit of testing more robust while we're here. Notice that is the less recemt tests I had been directly modifying the `AppConfig` to test the desired behavior -- we risk having side effects in our tests because of that. Instead we could be safely mocking the return values of a utility method to test the different cases.

Reply:
Moved to config-utils